README.md

```diff
@@ -29,24 +29,77 @@ the [ngx_lua module](http://wiki.nginx.org/HttpLuaModule), [LuaJIT 2.0](http://l
+`aws_credentials` config option specifies which provider to use. 
+
+If no `aws_credentials` is provided then the library will try to guess one:
+ 
+* if `aws_access_key` and `aws_secret_key` are provided then Basic Credentials Provider is used. 
+* Otherwise the IAM Credentials Provider is used. 
+   
+>INFO: This library supports the latest AWS V4 signature which means you can use any of the latest AWS APIs without any problem.
+
+
+
+#### Basic Credentials
+Basic credentials work with `secret_key` and `access_key`.
+```lua
+aws_credentials = {
+    provider = "api-gateway.aws.AWSBasicCredentials",
+    access_key = "replace-me",
+    secret_key = "replace-me"
+}
+```
+>INFO: For a better security inside the AWS environment use IAM or STS credentials.
+
+### IAM Credentials
+This is probably the most popular credentials provider to be used inside the AWS environment. 
+To learn more about IAM Credentials see [IAM Roles for Amazon EC2](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html).
+
+This credentials provider discovers automatically the `IAM Role` associated to the EC2 instance retrieving its credentials and caching them.
+  This is a more secure method for signing AWS requests as credentials are short lived and the NGINX configuration doesn't need to maintain any `access_key` or `secret_key` nor worry about rotating the keys. 
+
+```lua
+aws_credentials = {
+    provider = "api-gateway.aws.AWSIAMCredentials",
+    shared_cache_dict = "my_dict"    -- the name of a shared dictionary used for caching IAM Credentials
+}
+```
+
+### STS Credentials
+AWS Security Token Service(STS) provides a great way to get limited-privilege credentials for accessing AWS Services.
+ To learn more about STS Credentials see [Getting Temporary Credentials with STS](http://docs.aws.amazon.com/AWSSdkDocsJava/latest/DeveloperGuide/prog-services-sts.html) and the [STS](#sts) section bellow.
+```lua
+aws_credentials = {
+    provider = "api-gateway.aws.AWSSTSCredentials",
+    role_ARN = "arn:aws:iam::111111:role/assumed_role_kinesis", -- ARN of the role to assume
+    role_session_name = "kinesis-session",                      -- a name for this session
+    shared_cache_dict = "shared_cache"                          -- shared dict for caching the credentials
+}    
+```
+Unlike IAM Credentials that exposes a single IAM Role for each EC2 instance, STS Credentials allows an EC2 instance to assume multiple roles each with its own access policy.
+
+This credentials provider uses [SecurityTokenService](src/lua/api-gateway/aws/sts/SecurityTokenService.lua) for making requests to STS and SecurityTokenService uses the IAM Credentials provider for making the call.
+ It is strongly recommended to provide the `shared_cache_dict` in order to improve performance. The temporary credentials obtained from STS are stored in it for up to 60 minutes.
```

Comment:
The temporary credentials obtained from STS are stored in `shared_cache_dict` for up to 60 minutes.